generic/piranha/src/models/source_code_unit.rs

```diff
@@ -71,9 +77,31 @@ impl SourceCodeUnit {
       .expect("Could not generate new tree!");
     self.ast = new_tree;
     self.code = new_source_code;
+    // Handle errors, like removing extra comma.
+    self.handle_error(parser);
+    if self.ast.root_node().has_error() {
+      panic!("Produced syntactically incorrect source code {}", self.code());
+  }
     ts_edit
   }
 
+  /// Sometimes our rewrite rules may produce errors (recoverable errors), like failing to remove an extra comma.
+  /// This function, applies the recovery strategies.
+  /// Currently, we only support recovering from extra comma. 
+  fn handle_error(&mut self, parser: &mut Parser) {
+    if self.ast.root_node().has_error() {
```

Comment:
why is this named `handle_error`? Can we name it something more descriptive?

Reply:
Done.